File: Classes/User/Player/playerProfile.vb
''' <summary>
''' Get or Set the current Player data.
''' </summary>
Public Class playerProfile
	Private iniFileHandler As New iniHandler()
	Dim stringProfileData(,) As String
''' <summary>
''' Initializes Profile class.
''' </summary>
	Friend Sub New()
		
	End Sub
''' <summary>
''' Creates a new Profile.
''' </summary>
	Public Sub NewPlayerProfile(ProfileInfo() As String)
		stringProfileData = New String(,) {{"[Player Profile]", "<ProfileID>", ProfileInfo(0), "<NameFirst>", ProfileInfo(1), "<NameLast>", ProfileInfo(2), "<BirthYear>", ProfileInfo(3), "<BirthMonth>", ProfileInfo(4), "<BirthDay>", ProfileInfo(5), "<Gender>", ProfileInfo(6), "<E-Mail>", ProfileInfo(7), "<E-Mail Notification>", ProfileInfo(8)}}
		gamecache.currentSettings.LastProfile = ProfileInfo(0)
		'Start Writing New Profile
		SaveState()
	End Sub
''' <summary>
''' Loads a Profile file.
''' </summary>
	Friend Sub LoadPlayerProfile()
		stringProfileData = iniFilehandler.Load(System.IO.Directory.GetCurrentDirectory & "\save\" & gamecache.currentSettings.LastProfile & "\player\profile.ini")
	End Sub
''' <summary>
''' Searches in stringProfileData for requested Setting and set/returns that value.
''' </summary>
''' <param name="InformationRequest">Requested information as String. And saves when Set.</param>
''' <returns></returns>
	Public Property ReturnValue(InformationRequest As String) As String
		Get
			Dim integerColumn As Integer = 0
			Dim integerRow As Integer = 0
			Do
				Do
					If stringProfileData(integerColumn, integerRow).Contains(InformationRequest) Then
						Return stringProfileData(integerColumn, integerRow + 1)
						Exit Property
					End If
					integerRow += 1
				Loop Until integerRow > stringProfileData.GetUpperBound(1)
				integerColumn += 1
			Loop Until integerColumn > stringProfileData.GetUpperBound(0)
			Return ""
		End Get
		
		Set(Value As String)
			Dim integerColumn As Integer = 0
			Dim integerRow As Integer = 0
			Do
				Do
					If stringProfileData(integerColumn, integerRow).Contains(InformationRequest) Then
						stringProfileData(integerColumn, integerRow + 1) = Value
						SaveState()
						Exit Property
					End If
					integerRow += 1
				Loop Until integerRow > stringProfileData.GetUpperBound(1)
				integerColumn += 1
			Loop Until integerColumn > stringProfileData.GetUpperBound(0)
		End Set
	End Property
''' <summary>
''' Saves all changes made to the Profile data.
''' </summary>
	Private Sub SaveState()
		System.IO.Directory.CreateDirectory(System.IO.Directory.GetCurrentDirectory & "\save\" & gamecache.currentSettings.LastProfile & "\player")
		iniFilehandler.Save(System.IO.Directory.GetCurrentDirectory & "\save\" & gamecache.currentSettings.LastProfile & "\player\profile.ini", stringProfileData)
	End Sub
''' <summary>
''' Returns a String with the ProfileID.
''' </summary>
	Public Property ProfileID As String
		Get
			Return ReturnValue("ProfileID")
		End Get
		Set(Value As String)
			ReturnValue("ProfileID") = Value
			SaveState()
		End Set
	End Property
''' <summary>
''' Returns a String with the First name.
''' </summary>
	Public Property FirstName As String
		Get
			Return ReturnValue("NameFirst")
		End Get
		Set(Value As String)
			ReturnValue("NameFirst") = Value
			SaveState()
		End Set
	End Property
''' <summary>
''' Returns a String with the Last name.
''' </summary>
	Public Property LastName As String
		Get
			Return ReturnValue("NameLast")
		End Get
		Set(Value As String)
			ReturnValue("NameLast") = Value
			SaveState()
		End Set
	End Property
''' <summary>
''' Returns an Integer with the Birth Year.
''' </summary>
	Public Property BirthYear As Integer
		Get
			Return CType(ReturnValue("BirthYear"), Integer)
		End Get
		Set(Value As Integer)
			ReturnValue("BirthYear") = Value.ToString
			SaveState()
		End Set
	End Property
''' <summary>
''' Returns an Integer with the Birth Month.
''' </summary>
	Public Property BirthMonth As Integer
		Get
			Return CType(ReturnValue("BirthMonth"), Integer)
		End Get
		Set(Value As Integer)
			ReturnValue("BirthMonth") = Value.ToString
			SaveState()
		End Set
	End Property
''' <summary>
''' Returns an Integer with the Birth Day.
''' </summary>
	Public Property BirthDay As Integer
		Get
			Return CType(ReturnValue("BirthDay"), Integer)
		End Get
		Set(Value As Integer)
			ReturnValue("BirthDay") = Value.ToString
			SaveState()
		End Set
	End Property
''' <summary>
''' Returns/Sets an String with the Gender
''' </summary>
	Public Property Gender As String
		Get
			Return ReturnValue("Gender")
		End Get
		Set(Value As String)
			ReturnValue("Gender") = Value
			SaveState()
		End Set
	End Property
''' <summary>
''' Returns/Sets a String with the stringEMail.
''' </summary>
	Public Property EMail As String
		Get
			Return ReturnValue("EMail")
		End Get
		Set(Value As String)
			ReturnValue("EMail") = Value
			SaveState()
		End Set
	End Property
''' <summary>
''' Returns/Sets a True/False if subscribed to EMail Notification.
''' </summary>
	Public Property Notification As Boolean
		Get
			Return CType(ReturnValue("Notification"), Boolean)
		End Get
		Set(Value As Boolean)
			ReturnValue("Notification") = Value.ToString
			SaveState()
		End Set
	End Property
End Class

File: Game/Classes/User/Player/playerProfile.cs
using System;

/// <summary>
/// Get or Set the current Player data.
/// </summary>
public class playerProfile
{
    public string InformationRequest;
    iniHandler iniFileHandler = new iniHandler();
    string[,] stringProfileData;

    /// <summary>
    /// Creates a new Profile.
    /// </summary>
    public void NewPlayerProfile(string[] ProfileInfo)
    {
        stringProfileData = new string[,] {
            { "[Player Profile]", "" },
            { "<ProfileID>", ProfileInfo[0] },
            { "<NameFirst>", ProfileInfo[1] },
            { "<NameLast>", ProfileInfo[2] },
            { "<BirthYear>", ProfileInfo[3] },
            { "<BirthMonth>", ProfileInfo[4] },
            { "<BirthDay>", ProfileInfo[5] },
            { "<Gender>", ProfileInfo[6] },
            { "<E-Mail>", ProfileInfo[7] },
            { "<E-Mail Notification>", ProfileInfo[8] }
        };
        gamecache.currentSettings.LastProfile = ProfileInfo[0];
        //Start Writing New Profile
        SaveState();
    }
    
    /// <summary>
    /// Loads a Profile file.
    /// </summary>
    internal void LoadPlayerProfile()
    {
        stringProfileData = iniFileHandler.Load(System.IO.Directory.GetCurrentDirectory() + "\\save\\" + gamecache.currentSettings.LastProfile + "\\player\\profile.ini");
    }
    
    /// <summary>
    /// Searches in stringProfileData for requested Setting and set/returns that value.
    /// </summary>
    /// <returns></returns>
    public string ReturnValue {
        get {
            int integerColumn = 0;
            int integerRow = 0;
            do {
                do {
                    if (stringProfileData[integerColumn, integerRow].Contains(InformationRequest)) {
                        return stringProfileData[integerColumn, integerRow + 1];
                    }
                    integerRow += 1;
                } while (!(integerRow > stringProfileData.GetUpperBound(1)));
                integerColumn += 1;
            } while (!(integerColumn > stringProfileData.GetUpperBound(0)));
            return "";
        }

        set {
            int integerColumn = 0;
            int integerRow = 0;
            do {
                do {
                    if (stringProfileData[integerColumn, integerRow].Contains(InformationRequest)) {
                        stringProfileData[integerColumn, integerRow + 1] = value;
                        SaveState();
                        return;
                    }
                    integerRow += 1;
                } while (!(integerRow > stringProfileData.GetUpperBound(1)));
                integerColumn += 1;
            } while (!(integerColumn > stringProfileData.GetUpperBound(0)));
        }
    }

    /// <summary>
    /// Saves all changes made to the Profile data.
    /// </summary>
    private void SaveState()
    {
        System.IO.Directory.CreateDirectory(System.IO.Directory.GetCurrentDirectory() + "\\save\\" + gamecache.currentSettings.LastProfile + "\\player");
        iniFileHandler.Save(System.IO.Directory.GetCurrentDirectory() + "\\save\\" + gamecache.currentSettings.LastProfile + "\\player\\profile.ini", stringProfileData);
    }

    /// <summary>
    /// Returns a String with the ProfileID.
    /// </summary>
    public string ProfileID {
        get {
            InformationRequest = "ProfileID";
            return ReturnValue;
        }
        set {
            InformationRequest = "ProfileID";
            ReturnValue = value;
            SaveState();
        }
    }
    
    /// <summary>
    /// Returns a String with the First name.
    /// </summary>
    public string FirstName {
        get {
            InformationRequest = "NameFirst";
            return ReturnValue;
        }
        set {
            InformationRequest = "NameFirst";
            ReturnValue = value;
            SaveState();
        }
    }
    
    /// <summary>
    /// Returns a String with the Last name.
    /// </summary>
    public string LastName {
        get {
            InformationRequest = "NameLast";
            return ReturnValue;
        }
        set {
            InformationRequest = "NameLast";
            ReturnValue = value;
            SaveState();
        }
    }
    
    /// <summary>
    /// Returns an Integer with the Birth Year.
    /// </summary>
    public int BirthYear {
        get {
            InformationRequest = "BirthYear";
            return Convert.ToInt32(ReturnValue);
        }
        set {
            InformationRequest = "BirthYear";
            ReturnValue = value.ToString();
            SaveState();
        }
    }
    
    /// <summary>
    /// Returns an Integer with the Birth Month.
    /// </summary>
    public int BirthMonth {
        get {
            InformationRequest = "BirthMonth";
            return Convert.ToInt32(ReturnValue);
        }
        set {
            InformationRequest = "BirthMonth";
            ReturnValue = value.ToString();
            SaveState();
        }
    }
    
    /// <summary>
    /// Returns an Integer with the Birth Day.
    /// </summary>
    public int BirthDay {
        get {
            InformationRequest = "BirthDay";
            return Convert.ToInt32(ReturnValue);
        }
        set {
            InformationRequest = "BirthDay";
            ReturnValue = value.ToString();
            SaveState();
        }
    }
    
    /// <summary>
    /// Returns/Sets an String with the Gender
    /// </summary>
    public string Gender {
        get {
            InformationRequest = "Gender";
            return ReturnValue;
        }
        set {
            InformationRequest = "Gender";
            ReturnValue = value;
            SaveState();
        }
    }
    
    /// <summary>
    /// Returns/Sets a String with the stringEMail.
    /// </summary>
    public string EMail {
        get {
            InformationRequest = "EMail";
            return ReturnValue;
        }
        set {
            InformationRequest = "EMail";
            ReturnValue = value;
            SaveState();
        }
    }
    
    /// <summary>
    /// Returns/Sets a True/False if subscribed to EMail Notification.
    /// </summary>
    public bool Notification {
        get {
            InformationRequest = "Notification";
            return Convert.ToBoolean(ReturnValue);
        }
        set {
            InformationRequest = "Notification";
            ReturnValue = value.ToString();
            SaveState();
        }
    }
}